ngx_http_vod_module.c

```diff
@@ -3418,11 +3422,19 @@ ngx_http_vod_async_http_read(ngx_http_vod_http_reader_state_t *state, ngx_buf_t
 	child_params.range_start = offset;
 	child_params.range_end = offset + size;
 
+	upstream_location = &conf->upstream_location;
+
+	if (conf->request_handler == ngx_http_vod_mapped_request_handler &&
+		conf->remote_upstream_location.len != 0)
+	{
+		upstream_location = &conf->remote_upstream_location;
+	}
+
```

Comment:
if remote_upstream_location is set in a mapped location, it will apply to both mapping requests and mp4 requests, the idea was to be able to control these separately (remote_upstream_location will apply to mp4, and upstream_location  will apply to json)